overlord/snapstate/refreshhints.go

```diff
@@ -179,6 +179,20 @@ func refreshHintsFromCandidates(st *state.State, updates []*snap.Info, ignoreVal
 			continue
 		}
 
+		var registries []RegistryID
+		for _, plug := range update.Plugs {
+			if plug.Interface != "registry" {
+				continue
+			}
+
+			account, registry, _, err := snap.RegistryPlugAttrs(plug)
+			if err != nil {
+				return nil, err
+			}
+
+			registries = append(registries, RegistryID{Account: account, Registry: registry})
+		}
```

Comment:
should this code be a shared helper or something? like the code in target and snapstate? but maybe it's easier after the Andrew refactor

Reply:
There are now (after merging with the refactor) only two of these loops and they're slightly different as one takes a map. I could convert that into a slice and reuse the same code but I don't think the diff will improve much. Maybe if we end up adding more of these?